lib/doctest.js

```diff
@@ -492,10 +525,14 @@ const functionEval = async source => {
 
 const log = results => {
   console.log (
-    results.reduce ((s, [correct]) => `${s}${correct ? '.' : 'x'}`, '')
+    results.reduce ((s, {correct}) => `${s}${correct ? '.' : 'x'}`, '')
   );
-  results.forEach (([correct, actual, expected, line]) => {
+  results.forEach (({correct, input, output}) => {
     if (!correct) {
+      const format = r => r.throws ? `! ${r.exception}` : show (r.result);
```

Comment:
....is it safe to naively cast it to string here?

Reply:
I believe this to be equivalent to ``io['!'] ? `! ${expected}` : show (expected)`` in the existing code.

Reply:
True, so it's not a regression at least.